src/components/data-table/ha-data-table.ts

```diff
@@ -571,6 +571,18 @@ export class HaDataTable extends BaseElement {
         width: 24px;
       }
 
+      .mdc-data-table__header-cell--icon {
+        text-align: center;
+      }
+
+      .mdc-data-table__header-cell--icon:first-child {
+        padding-left: 24px;
```

Comment:
Does this also work when the table contains a 40px wide state badge?

Reply:
Yes: 
![image](https://user-images.githubusercontent.com/5662298/75567122-47b33880-5a51-11ea-9ec3-cc5a20f1df01.png)